src/Vault.sol

```diff
@@ -516,13 +527,20 @@ contract Vault is IERC4626, ERC20Permit, ILiquidationSource, Ownable {
     uint8 _v,
     bytes32 _r,
     bytes32 _s
-  ) external returns (uint256) {
+  ) external onlyVaultCollateralized returns (uint256) {
     _permit(IERC20Permit(address(_asset)), _owner, address(this), _assets, _deadline, _v, _r, _s);
     return _depositAssets(_assets, _owner, _owner);
   }
 
   /// @inheritdoc IERC4626
-  function mint(uint256 _shares, address _receiver) external virtual override returns (uint256) {
+  function mint(
+    uint256 _shares,
+    address _receiver
+  ) external virtual override onlyVaultCollateralized returns (uint256) {
+    if (_shares > _maxMint()) {
+      revert MintMoreThanMax(_receiver, _shares, _maxMint());
+    }
```

Comment:
Similar to above comment, this function is technically checking collateralization twice since it's also checked in `_maxMint()`

Reply:
Fixed in this commit: https://github.com/GenerationSoftware/pt-v5-vault/pull/51/commits/dda6ac44c4f7787dce9c1e75475b0f21a51e2cb0